questionnaire/questionnaire-service.js

```diff
@@ -117,18 +117,20 @@ function createQuestionnaireService({
                 !submissionResponse.body ||
                 submissionResponse.body !== 'Message sent'
             ) {
-                await updateQuestionnaireSubmissionStatus(questionnaireId, 'FAILED');
-                const slackService = createSlackService();
-                slackService.sendMessage({
-                    appReference: `${process.env.APP_ENV || 'dev'}.reporter.webhook`,
-                    messageBodyId: 'message-bus-down',
-                    templateParameters: {
-                        timeStamp: new Date().getTime()
-                    }
-                });
+                throw new VError(
+                    `Questionnaire with id "${questionnaireId}" was not successfully added to the Message Bus queue`
+                );
```

Comment:
In the interest of best practice, should this function not return something, or perhaps be structured better? It awaits a response to the post, errors if it doesn't like the result, otherwise nothing?

Reply:
what do you suggest it should return?

Reply:
`submissionResponse` should probably be returned, as it fulfils the main purpose of this function.

Reply:
That is the only useful that could be returned but I don't see much of a point in doing that other than to adhere to the standard of having an explicit return. This functions purpose is to kick off a submission or throw and fail. Returning anything to the caller won't help here (as it stands).